tests/menu_items_registry.test.ts

```diff
@@ -174,17 +174,42 @@ describe("Menu Item actions", () => {
     expect(dispatch).toHaveBeenCalledWith("PASTE_FROM_OS_CLIPBOARD", {
       target: target("A1"),
       text,
+      pasteOption: "onlyValue",
     });
   });
 
-  test("Edit -> paste_special -> paste_special_format", () => {
+  test("Edit -> paste_special -> paste_special_format", async () => {
+    const text = "in OS clipboard";
+    await env.clipboard.writeText(text);
     doAction(["edit", "paste_special", "paste_special_format"], env);
-    expect(dispatch).toHaveBeenCalledWith("PASTE", {
-      target: env.model.getters.getSelectedZones(),
+    await nextTick();
+    expect(dispatch).toHaveBeenCalledWith("PASTE_FROM_OS_CLIPBOARD", {
+      target: target("A1"),
+      text,
       pasteOption: "onlyFormat",
     });
   });
 
+  test("'Edit -> paste' after altering content", async () => {
+    const text = "in OS clipboard";
+    await env.clipboard.writeText(text);
+    const content = env.model.getters.getClipboardTextContent();
+    doAction(["edit", "paste"], env);
+    await nextTick();
+    if (text !== content) {
+      expect(dispatch).toHaveBeenCalledWith("PASTE_FROM_OS_CLIPBOARD", {
+        target: target("A1"),
+        text,
+        pasteOption: undefined,
+      });
+    } else {
+      expect(dispatch).toHaveBeenCalledWith("PASTE", {
+        target: target("A1"),
+        pasteOption: undefined,
+      });
+    }
```

Comment:
The `if/else` have nothing to do here, you are writing a test so you are controlling whether test === content or not. As stated in my previous comment, you should have 2 separate tests : one for PASTE_FROm_OS and one for PASTE.